functions/Restore-DbaBackup.ps1

```diff
@@ -0,0 +1,215 @@
+function Restore-DbaBackup
+{
+<#
+.SYNOPSIS 
+Restores a SQL Server Database from a set of backupfiles
+
+.DESCRIPTION
+Scans a given folder for Full, Differential and Log backups. 
+OR
+Takes a set of folder arrays and processes them for Full, Differential and Log backups.
+
+These are then filtered and restored to a specified SQL Server intance and file location
+
+The backup LSN chain and RecoveryForkID will also be checked to ensure the restore is valid
+
+It can also generate restore scripts, both as part of a restore or as it's only action
+
+.PARAMETER SqlServer
+The SQL Server instance. 
+
+.PARAMETER SqlCredential
+Allows you to login to servers using SQL Logins as opposed to Windows Auth/Integrated/Trusted. 
+
+.PARAMETER Path
+Path to SQL Server backup file. This can be a full, differential or log backup file.
+
+.PARAMETER Files
+A Files object(s) containing SQL Server backup files. 
+
+.PARAMETER RestoreLocation
+Path to restore the SQL Server backups to on the target inance
+
+.PARAMETER RestoreTime
+Specify a DateTime object to which you want the database restored to. Default is to the latest point available 
+
+.PARAMETER OlaStyle
```

Comment:
It's used a shortcut further down. If it finds an empty FULL folder then it doesn't bother going any further. Any other folder structure and we have to scan all .baks to see if they are diffs or fulls

MaintenanceSolutionBackup works for me.